src/navigation/navigationTabs/tabs.jsx

```diff
@@ -0,0 +1,572 @@
+            tabsVisible: sortedVisibleTabs,
+        };
+
+        return reducedItems;
+    }
+
+    getTabProps(tab) {
+        const {
+            className,
+            collapsed,
+            disabled,
+            key,
+            onClick,
+            selected,
+            tabIndex,
+            title,
+        } = tab;
+
+        return {
+            children: title,
+            classNames: this.getClassNamesFor('tab', {
+                className,
+                collapsed,
+                disabled,
+                selected,
+                tabIndex,
+            }),
+            id: PREFIX_TAB + key,
+            key: PREFIX_TAB + key,
+            originalKey: key,
+            onClick,
+            onChange: this.onChangeTab,
+            ref: (ref) => { this.tabRefs[PREFIX_TAB + key] = ref; },
+            selected,
+        };
+    }
+
+    getPanelProps(panel, isHidden) {
+        const {
+            className,
+            content,
+            getContent,
+            key,
+        } = panel;
+
+        return {
+            getContent,
+            children: content,
+            key: PANEL_PREFIX + key,
+            id: PANEL_PREFIX + key,
+            tabId: PREFIX_TAB + key,
+            classNames: this.getClassNamesFor('panel', { className, isHidden }),
+            isHidden,
+        };
+    }
+
+    getClassNamesFor(type, itemStates) {
+        const {
+            className = '',
+            disabled,
+            selected,
+            tabIndex,
+        } = itemStates;
+
+        switch (type) {
+            case 'tab':
+                return Classnames(
+                    `${BEM_NAVIGATION_TAB_ROOT_CLASS}`,
+                    className,
+                    {
+                        [`${BEM_NAVIGATION_TAB_ROOT_CLASS}--first`]: !tabIndex,
+                        [`${BEM_NAVIGATION_TAB_ROOT_CLASS}--selected`]: selected,
+                        [`${BEM_NAVIGATION_TAB_ROOT_CLASS}--disabled`]: disabled,
+                    },
+                );
+            case 'panel':
+                return Classnames(
+                    `${BEM_NAVIGATION_TABS}--panel-content`,
+                    className,
+                );
+            default:
+                return '';
+        }
+    }
+
+    getSelectedTabKey() {
+        const {
+            items,
+        } = this.props;
+
+        const {
+            selectedTabKey,
+        } = this.state;
+
+        if (!selectedTabKey) {
+            if (!items[0]) {
+                return undefined;
+            }
+
+            return items[0].key;
+        }
+
+        return selectedTabKey;
+    }
+
+    getExpandedTabs(panels, selectedTabKey) {
+        if (panels[selectedTabKey]) {
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            return <TabPanel {...this.getPanelProps(panels[selectedTabKey])} />;
+        }
+
+        return null;
+    }
+
+    render() {
+        const {
+            classes,
```

Comment:
How do we include the classes inside the PropTypes validation without exposing in the DOCS? Or It doesn't matter it appears even when is not a prop the user can pass to the component?

Reply:
It is okay that `classes` prop is exposed in the documentation.  And my understanding is that component consumers _can_ in fact pass their own class names using that prop and as well, for further customization if needed.  @morethanfire perhaps you can elaborate.

Reply:
Yeah, exactly, it is good that it is exposed.